R/infix_spaces_linter.R

```diff
@@ -144,11 +144,11 @@ infix_spaces_linter <- function(exclude_operators = NULL, allow_multiple_spaces
 
   # NB: preceding-sibling::* and not preceding-sibling::expr because
   #   of the foo(a=1) case, where the tree is <SYMBOL_SUB><EQ_SUB><expr>
-  # NB: position() > 1 for the unary case, e.g. x[-1]
+  # NB: parent::*[count(expr) + count(OP-LEFT-PAREN) > 1] for the unary case, e.g. x[-1]
+  #  OP-LEFT-PAREN for EQ_SUB case with missing argument like alist(a =)
   # NB: the last not() disables lints inside box::use() declarations
-  xpath <- glue::glue("//*[
-    ({xp_or(paste0('self::', infix_tokens))})
-    and position() > 1
+  xpath <- paste(collapse = "|", glue::glue("//{infix_tokens}[
+    parent::*[count(expr) + count(SYMBOL_SUB) > 1]
```

Comment:
is `count(expr | SYMBOL_SUB)` a thing?

Reply:
Nice! I still don't have a good sense of when `|` can be used like this. It passed tests locally, so let's go for it!